### ISSUE TITLE
[BUG] Failed to run AutoMM in Multiclass task

### ISSUE THREAD

**Issue by rsj123** [autogluon/autogluon#3414]
**Bug Report Checklist**

- [x] I provided code that demonstrates a minimal reproducible example.  
- [x] I confirmed bug exists on the latest mainline of AutoGluon via source install.  
- [ ] I confirmed bug exists on the latest stable version of AutoGluon.  

**Describe the bug**

>RuntimeError: Predictions and targets are expected to have the same shape, but got torch.Size([4, 2]) and torch.Size([4]).

**To Reproduce**

``` python
import pandas as pd
from autogluon.tabular import FeatureMetadata
from autogluon.tabular.configs.hyperparameter_configs import get_hyperparameter_config
from autogluon.tabular import TabularPredictor
from autogluon.multimodal import MultiModalPredictor
import torch
import numpy as np

torch.set_float32_matmul_precision('high')
np.random.seed(42)
eval_metric = "f1"


train_data = pd.read_csv("label_1.csv")
label = "label"

predictor = MultiModalPredictor(label=label,eval_metric=eval_metric).fit(
    train_data=train_data,
    presets='multilingual',
    time_limit=3600,
)
```

**Screenshots / Logs**

https://gist.github.com/rsj123/ec952ed316a2e512a93d9ddb1ce1ad00

**Installed Versions**

<details>

INSTALLED VERSIONS
------------------
date                   : 2023-07-22
time                   : 00:00:31.354196
python                 : 3.10.8.final.0
OS                     : Linux
OS-release             : 5.4.0-153-generic
Version                : #170-Ubuntu SMP Fri Jun 16 13:43:31 UTC 2023
machine                : x86_64
processor              : x86_64
num_cores              : 128
cpu_ram_mb             : 515688
cuda version           : 12.535.54.03
num_gpus               : 1
gpu_ram_mb             : [23336]
avail_disk_size_mb     : 223055

accelerate             : 0.16.0
autogluon              : 0.8.3b20230720
autogluon.common       : 0.8.3b20230720
autogluon.core         : 0.8.3b20230720
autogluon.features     : 0.8.3b20230720
autogluon.multimodal   : 0.8.3b20230720
autogluon.tabular      : 0.8.3b20230720
autogluon.timeseries   : 0.8.3b20230720
boto3                  : 1.28.3
catboost               : 1.2
defusedxml             : 0.7.1
evaluate               : 0.3.0
fastai                 : 2.7.12
gluonts                : 0.13.2
grpcio                 : 1.50.0
hyperopt               : 0.2.7
imodels                : None
jinja2                 : 3.1.2
joblib                 : 1.3.1
jsonschema             : 4.17.3
lightgbm               : 3.3.5
matplotlib             : 3.6.3
mlforecast             : 0.7.3
networkx               : 3.1
nlpaug                 : 1.1.11
nltk                   : 3.8.1
nptyping               : 2.4.1
numpy                  : 1.24.1
omegaconf              : 2.2.3
onnxruntime-gpu        : None
openmim                : 0.3.9
pandas                 : 1.5.3
Pillow                 : 9.4.0
psutil                 : 5.9.4
pydantic               : 1.10.11
PyMuPDF                : None
pytesseract            : 0.3.10
pytorch-lightning      : 1.9.5
pytorch-metric-learning: 1.7.3
ray                    : 2.3.1
requests               : 2.28.1
scikit-image           : 0.19.3
scikit-learn           : 1.2.2
scikit-learn-intelex   : None
scipy                  : 1.11.1
seqeval                : 1.2.2
setuptools             : 68.0.0
skl2onnx               : None
statsforecast          : 1.4.0
statsmodels            : 0.14.0
tabpfn                 : None
tensorboard            : 2.11.2
text-unidecode         : 1.3
timm                   : 0.9.2
torch                  : 1.13.1+cu117
torchmetrics           : 0.11.4
torchvision            : 0.14.1+cu117
tqdm                   : 4.64.1
transformers           : 4.26.1
ujson                  : 5.8.0
vowpalwabbit           : None
xgboost                : 1.7.6

</details>

**Comment by rsj123** [comment]
dataset:
https://gist.github.com/rsj123/f1505ff06fe2940435fa8f04babeb90c

**Comment by rsj123** [comment]
After testing I'm guessing it has something to do with the f1 metrics.
It work fine with acc and roc_auc with the same dataset.

**Comment by Crispy13** [comment]
same here

**Comment by Crispy13** [comment]
In my case, 'f1_macro' instead of 'f1' works.

**Comment by xiechenhao** [comment]
same here

**Comment by vnghia** [comment]
I think this bug can be fixed by replacing `(torchmetrics.classification.BinaryAUROC, torchmetrics.classification.BinaryAveragePrecision)` to `(torchmetrics.classification.BinaryAUROC, torchmetrics.classification.BinaryAveragePrecision, torchmetrics.classification.BinaryF1Score)` in this line https://github.com/autogluon/autogluon/blob/a24b80e5ea476527f286dc3b0369b1b165567930/multimodal/src/autogluon/multimodal/optimization/lit_module.py#L203

I can send a PR but not sure why no one has looked into this issue yet :/

**Comment by tonyhoo** [comment]
Thanks for reporting the issue and we might be able to incorporate the fix into [PR](https://github.com/autogluon/autogluon/pull/3650/)

**Comment by zhiqiangdon** [comment]
@rsj123 We didn't test f1 metric much. Could you do some tests and submit a PR? I think our `evaluate` API only support binary task for now. See https://github.com/autogluon/autogluon/blob/master/multimodal/src/autogluon/multimodal/utils/metric.py#L215-L216. You can modify here accordingly to support both binary and multiclass.

**Comment by Crispy13** [comment]
My task is binary classification but I could not use 'f1' due to the same error. 
The label column in my data has only 0 or 1 values.

Did I do something wrong?

**Comment by FANGAreNotGnu** [comment]
F1_MACRO = "f1_macro", F1_MICRO = "f1_micro" and F1_WEIGHTED = "f1_weighted" are supported as both evaluation and validation metrics in PR https://github.com/autogluon/autogluon/pull/3696. Also the logic to infer evaluation/validation metric is updated.

**Comment by FANGAreNotGnu** [comment]
Closing the issue as it's been solved in #3822
